examples/medplum-scheduling-demo/src/pages/SchedulePage.tsx

```diff
@@ -9,13 +9,40 @@ import 'react-big-calendar/lib/css/react-big-calendar.css';
 import { useNavigate } from 'react-router-dom';
 import { CreateUpdateSlot } from '../components/CreateUpdateSlot';
 import { ScheduleContext } from '../Schedule.context';
+import { Title } from '@mantine/core';
+import { CreateAppointment } from '../components/CreateAppointment';
+
+interface OnClickEventModalProps {
+  event: Event | undefined;
+  readonly opened: boolean;
+  readonly handlers: {
+    readonly open: () => void;
+    readonly close: () => void;
+    readonly toggle: () => void;
+  };
+}
+
+// This helper component manage the modal to be shown when an event is clicked
+function OnClickEventModal(props: OnClickEventModalProps): JSX.Element {
```

Comment:
Needs a clearer name

Reply:
Yes, I agree, I didn't like that name either. I'll refactor it.